### PR TITLE
feat(sequencer-utils): add `--chain-id` to sequencer genesis utils

### PR DESCRIPTION
**Pull request by noot**
## Summary
utils to set the chain ID in the cometbft genesis.json file.

## Background
need it for ibc testing (we need to set the chain id when starting hermes) this makes my life a lot easier, cause otherwise it picks a random chain ID.

## Changes
- add `--chain-id` to sequencer genesis utils which sets the `chain_id` field in the cometbft genesis file to the given string

## Testing
unit tests + i ran it (`just run-cometbft` and `cat ~/.cometbft/config/genesis.json`)

### CODE REVIEW

File: crates/astria-sequencer-utils/src/genesis_parser.rs [hunk 1]
```diff
@@ -60,11 +64,12 @@ impl GenesisParser {
     }
 }
 
-fn insert_app_state(dst: &mut Value, app_state: &Value) {
+fn insert_app_state_and_chain_id(dst: &mut Value, app_state: &Value, chain_id: String) {
```
Comment (SuperFluffy) [inline crates/astria-sequencer-utils/src/genesis_parser.rs:67]: This doesn't really scale IMO. Better to replace this by
```rust
dst["app_state"] = json!(source_genesis_data);
dst["chain_id"] = json!(chain_id);
```

**Review by sambukowski — APPROVED** [review]

File: crates/astria-sequencer-utils/src/genesis_parser.rs (hunk 1 above)
Comment (noot) [reply crates/astria-sequencer-utils/src/genesis_parser.rs:67]: like replace the whole function? that is harder for unit tests :/

Comment (SuperFluffy) [reply crates/astria-sequencer-utils/src/genesis_parser.rs:67]: Yeah, this should be rewritten so that not everything is stuffed into a single function.

**Review by SuperFluffy — APPROVED** [review]
unhappy with how this tool looks right now, but makes sense to not block it on a refactor.

**Pull request merged**